### PR TITLE
refactored main.py, added optional summarization, and option to modif…

### PR DESCRIPTION
**Pull request by Sommos**
refactored main.py with comments and consistent style 
- added optional summarization
- added option to modify everything vs file by file 
- remade .gitignore

### CODE REVIEW

**Comment by jawerty** [comment]
Awesome thanks @Sommos I'm gonna take a look at this tomorrow and review.

File: main.py [hunk 1]
```diff
@@ -183,92 +194,122 @@ def parse_scaffolding_result(output):
...
-    code_files = initiate_code_modification(code_files, modification_ask)
-
-  # dependency resolving
+    if modify_all:
+      # update all code blocks with a prediction using the modification ask of the user
+      code_files = initiate_code_modification(code_files, modification_ask)
+    else:
+      # update each related code block with a prediction using the modification ask of the user
+      for code_file in code_files:
+        if modification_ask in code_file["code_block"]:
```
Comment (jawerty) [inline main.py]: So far the comments are solid for cleaning it up. But not fully sure what this part is doing? This is checking if the modification ask is a substring in the code? Might be better to have some way to detect whether the code matches the modification ask. Could maybe vectorize the code and find matches.

Comment (Sommos) [reply main.py]: Sorry, just woke up myself (on UK time so am a bit delayed in responding). 

So yeah, it checks if it is a substring in the code. I'm not a fan of this way of doing it, as nested loops like this make it look clunky and feel unnecessary. Vectorizing the code to find matches is probably the better way of doing it. I've just put in a push to my branch that includes a rudimentary solution to vectorizing & finding if the code block matches the modification ask using 2 new functions:

- `vectorize_code` <-- This should vectorize the code block, and return a vector representation of the code block itself
- `find_closest_matches`  <-- This should find the closest matches to the modification prompt, and then return the indices of the closest matches to be used in the reworked `dev_loop` function

I'm only a junior, so still learning and this implementation might not be the best way of doing it!

Comment (jawerty) [reply main.py]: I do like the creativity in your vectorization implementation! Good job.

However, one thing is we want to use sentence transformers models for the vectorization here. And make it so people can get embeddings with any transformers model they'd like. Honestly I have to think through that feature more. Do you mind removing that part with the modification ask work/vectorization and I'll approve. I like the optional summarization and the clean up part you did

Comment (Sommos) [reply main.py]: Removed it, and pushed. It should be ready for merge now.

**Review by jawerty — APPROVED** [review]
LGTM

**Pull request merged**